### PR TITLE
Add routing perf test and fix link cost check

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: controller/network/debug.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package network
```
Comment (dovholuknf) [inline controller/network/debug.go:1]: needs header and a comment as to what this is all about. did you commit this intentionally? feels out of place - thus maybe a comment?

File: controller/network/path.go [hunk 2]
```diff
@@ -234,5 +234,6 @@ func minCost(q map[*Router]bool, dist map[*Router]int64) *Router {
 			min = d
 		}
 	}
+	//Debugf("next router: %v, cost: %v, unvisited: %v\n", selected.Id, min, len(q))
```
Comment (dovholuknf) [inline controller/network/path.go:237]: dead code should just be deleted?

File: controller/network/route_perf_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
+package network
```
Comment (dovholuknf) [inline controller/network/route_perf_test.go:1]: add copyright header pls

File: controller/network/route_perf_test.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+	var routerIdx int
+	addRouter := func() *Router {
+		router := newRouterForTest(fmt.Sprintf("router-%03d", routerIdx), "", transportAddr, nil)
+		network.Routers.markConnected(router)
+		routerIdx++
+		return router
+	}
+
+	for i := 0; i < 50; i++ {
```
Comment (dovholuknf) [inline controller/network/route_perf_test.go:41]: this feels like a function you'll use numerous times when testing. forgot to make that comment on the other pr. this seems like a util test function type stuff, no?

File: controller/network/route_perf_test.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+	}
+
+	b.StartTimer()
+	srcRouter := routers[0]
+	replaceIdx := len(routers) - 1
+	for i := 1; i < len(routers); i++ {
+		p, c, err := network.shortestPath(srcRouter, routers[i])
+		ctx.NoError(err)
+
+		//fmt.Printf("{cost: %v, path:[]string{", c)
```
Comment (dovholuknf) [inline controller/network/route_perf_test.go:145]: note: dead code

File: controller/network/route_perf_test.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+		{cost: 155, path: []string{"router-000", "router-056"}},
+		{cost: 280, path: []string{"router-000", "router-002", "router-055"}},
+		{cost: 198, path: []string{"router-000", "router-002", "router-054"}},
+		{cost: 308, path: []string{"router-000", "router-002", "router-053"}},
+		{cost: 273, path: []string{"router-000", "router-052"}},
+		{cost: 133, path: []string{"router-000", "router-051"}},
+		{cost: 141, path: []string{"router-000", "router-050"}},
+	}
+
+	b.StartTimer()
```
Comment (dovholuknf) [inline controller/network/route_perf_test.go:138]: just a comment: maybe StartTimer should "stop" the timer too? i keep seeing the 'StopTimer' function call at the top of these tests which I don't quite 'get why' other than just making sure that the setup bits aren't swept into the timer?

**Review by dovholuknf — APPROVED** [review]

File: controller/network/route_perf_test.go (hunk 6 above)
Comment (plorenz) [reply controller/network/route_perf_test.go:138]: It's just how go benchmark stuff works. By default the whole function is timed, but you can stop the timer and restart when you're done with expensive initialization.

File: controller/network/path.go (hunk 2 above)
Comment (plorenz) [reply controller/network/path.go]: fixed

File: controller/network/route_perf_test.go (hunk 3 above)
Comment (plorenz) [reply controller/network/route_perf_test.go:17]: fixed

**Review by dovholuknf — APPROVED** [review]

File: controller/network/route_perf_test.go (hunk 4 above)
Comment (plorenz) [reply controller/network/route_perf_test.go:46]: extracted these out, added to a test helper, put in util_test with stuff from debug.go

File: controller/network/route_perf_test.go (hunk 5 above)
Comment (plorenz) [reply controller/network/route_perf_test.go]: Fixed

File: controller/network/debug.go (hunk 1 above)
Comment (plorenz) [reply controller/network/debug.go]: Renamed to util_test and added a comment. Added some other potentially common test bits there as well

**Pull request merged**